### PR TITLE
Updated license terms docs based on content from ckeditor.com

### PR DESCRIPTION
**Pull request by Reinmar**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Docs: Updated license terms docs based on content from ckeditor.com.

---

### Additional information

I de-title-cased two titles.

### CODE REVIEW

**Comment by Reinmar** [comment]
image

File: docs/getting-started/licensing/license-and-legal.md [hunk 1]
```diff
@@ -1,27 +1,44 @@
...
+meta-description: Choosing the right license type for you needs.
 menu-title: Editor license
 order: 30
 modified_at: 2024-11-20
 ---
 
 # Editor license and legal terms
 
-The following legal notices apply to CKEditor&nbsp;5 and all software from CKEditor&nbsp;5 Ecosystem included with it.
+CKEditor is dual-licensed. This means we make it available under a choice between an open source and a commercial license.
```
Comment (Witoso) [inline docs/getting-started/licensing/license-and-legal.md:12]: `under a choice` sounds a bit clunky, but I may be wrong, @godai78 could you check?

File: docs/getting-started/licensing/license-and-legal.md [hunk 2]
```diff
@@ -1,27 +1,44 @@
...
+
+* No GPL restrictions
+* Self-hosted CKEditor
+* Additionally paid Premium features such as:
+	* Asynchronous and Real-Time Collaboration
+	* Import from Word
+	* Export to Word
+	* Export to PDF
+	* Multi-level Lists
+	* Advanced Productivity
```
Comment (Witoso) [inline docs/getting-started/licensing/license-and-legal.md:34]: Advanced productivity is not a name of our capability, where did it come from?

File: docs/getting-started/licensing/license-and-legal.md [hunk 3]
```diff
@@ -1,27 +1,44 @@
...
 # Editor license and legal terms
 
-The following legal notices apply to CKEditor&nbsp;5 and all software from CKEditor&nbsp;5 Ecosystem included with it.
+CKEditor is dual-licensed. This means we make it available under a choice between an open source and a commercial license.
 
-Copyright (c) 2003–2024, CKSource Holding sp. z o.o. All rights reserved.
+## Open source license
 
-Licensed under the terms of [GNU General Public License Version 2 or later](http://www.gnu.org/licenses/old-licenses/gpl-2.0.html).
+The open source distribution of CKEditor is licensed under the terms of GPL 2+. If you plan to use CKEditor under this license, you can register your application using [the CKEditor GPL 2+ license registration form](https://info.ckeditor.com/l/1018062/2024-08-22/39jtg).
```
Comment (Witoso) [inline docs/getting-started/licensing/license-and-legal.md:16]: A couple of times LanguageTool marked for me GPL license as `This phrase is redundant (‘L’ stands for ‘license’). Use simply “GPL”.` But I think it's quite common to use it this way.

**Review by Witoso — DISMISSED** [review]
Just a few non-critical remarks/nitpicks, otherwise OK!

File: docs/getting-started/licensing/license-and-legal.md (hunk 1 above)
Comment (Reinmar) [reply docs/getting-started/licensing/license-and-legal.md:12]: FYI: This is straight from ckeditor.com new licensing page.

File: docs/getting-started/licensing/license-and-legal.md (hunk 2 above)
Comment (Witoso) [reply docs/getting-started/licensing/license-and-legal.md:34]: Ehh....

**Review by godai78 — DISMISSED** [review]
But I'd still advise changing the under choice phrase.

File: docs/getting-started/licensing/license-and-legal.md (hunk 1 above)
Comment (Reinmar) [reply docs/getting-started/licensing/license-and-legal.md:12]: Please open a ticket for ckeditor.com too.

**Review by godai78 — APPROVED** [review]

**Pull request merged**